nutanix/resource_nutanix_ndb_time_machine_cluster.go

```diff
@@ -0,0 +1,232 @@
+		CreateContext: resourceNutanixNDBTmsClusterCreate,
+		ReadContext:   resourceNutanixNDBTmsClusterRead,
+		UpdateContext: resourceNutanixNDBTmsClusterUpdate,
+		DeleteContext: resourceNutanixNDBTmsClusterDelete,
+		Schema: map[string]*schema.Schema{
+			"time_machine_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"nx_cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"sla_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "OTHER",
+			},
+			// computed
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"schedule_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"owner_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"source_clusters": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"log_drive_status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"date_created": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"date_modified": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"log_drive_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"source": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceNutanixNDBTmsClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).Era
+
+	req := &era.TmsClusterIntentInput{}
+
+	tmsID := d.Get("time_machine_id")
+
+	if nxcls, ok := d.GetOk("nx_cluster_id"); ok {
+		req.NxClusterID = utils.StringPtr(nxcls.(string))
+	}
+
+	if slaid, ok := d.GetOk("sla_id"); ok {
+		req.SLAID = utils.StringPtr(slaid.(string))
+	}
+
+	if clsType, ok := d.GetOk("type"); ok {
+		req.Type = utils.StringPtr(clsType.(string))
+	}
+
+	_, err := conn.Service.CreateTimeMachineCluster(ctx, tmsID.(string), req)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	uuid, er := uuid.GenerateUUID()
+
+	if er != nil {
+		return diag.Errorf("Error generating UUID for era clusters: %+v", err)
+	}
+	d.SetId(uuid)
+	log.Printf("NDB Time Machine Cluster with %s id is created successfully", d.Id())
+	return resourceNutanixNDBTmsClusterRead(ctx, d, meta)
+}
+
+func resourceNutanixNDBTmsClusterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).Era
+
+	tmsID := d.Get("time_machine_id")
+	clsID := d.Get("nx_cluster_id")
+	resp, err := conn.Service.ReadTimeMachineCluster(ctx, tmsID.(string), clsID.(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if err = d.Set("log_drive_id", resp.LogDrive); err != nil {
+		return diag.Errorf("error setting log_drive_id for tms clusters %s: %s", d.Id(), err)
```

Comment:
got following error while setting  <attr> for time machine cluster with id: %s

OR 

error occurred while setting  <attr> for time machine cluster with id: %s

Reply:
Make changes everywhere.

Reply:
sure.